server/rest/src/main/java/org/infinispan/rest/cachemanager/RestCacheManager.java

```diff
@@ -226,7 +226,10 @@ public CompletionStage<Collection<CacheDistributionInfo>> cacheDistribution(Stri
          }
 
          distributions.putIfAbsent(address, info);
-         if (distributions.size() == members.size() && !future.isCompletedExceptionally()) {
+      }).whenComplete((v, t) -> {
+         if (t != null) {
+            future.completeExceptionally(t);
```

Comment:
Same thing, this should rethrow the exception instead of setting a future and return the stage returned from `thenApply`